web/backend/test/routes/root.test.ts

```diff
@@ -119,27 +118,49 @@ test('runtime is running', async (t) => {
   assert.strictEqual(serviceInvalidOpenapi.statusCode, 500, 'service OpenAPI endpoint')
   assert.strictEqual(typeof serviceInvalidOpenapi.json().code, 'string')
 
-  const logs = await server.inject({
-    url: `/runtimes/${runtimePid}/logs`
-  })
-  assert.strictEqual(logs.statusCode, 200)
-
-  const result = logs.json<Log[]>()
-  assert.ok(result.some(({ msg }) => msg.includes('Starting the service')))
-  assert.ok(result.some(({ msg }) => msg.includes('Started the service')))
-  assert.ok(result.some(({ msg }) => msg.includes('Server listening at')))
-  assert.ok(result.some(({ msg }) => msg.includes('Platformatic is now listening')))
-
-  const [{ level, time, pid, hostname }] = result
-  assert.ok(typeof level, 'number')
-  assert.ok(typeof time, 'number')
-  assert.ok(typeof pid, 'number')
-  assert.ok(typeof hostname, 'string')
-
   const restart = await server.inject({
     method: 'POST',
-    url: `/runtimes/${pid}/restart`,
+    url: `/runtimes/${runtimePid}/restart`,
     body: {}
   })
   assert.strictEqual(restart.statusCode, 200, 'check for restart endpoint')
 })
+
+test('runtime logs websocket', async (t) => {
+  const port = await startWatt(t)
+  const server = await getServer(t)
+
+  const res = await server.inject({
+    url: '/runtimes?includeAdmin=true'
+  })
+  assert.strictEqual(res.statusCode, 200, 'runtimes endpoint')
+  const [runtime] = res.json()
+  const runtimePid = runtime.pid
+
+  const WebSocket = require('ws')
+  const ws = new WebSocket(`ws://localhost:${port}/runtimes/${runtimePid}/logs/ws`)
+
+  await new Promise((resolve, reject) => {
+    const timeout = setTimeout(() => {
+      reject(new Error('WebSocket connection timed out'))
+    }, 3000)
+
+    ws.on('open', () => {
+      clearTimeout(timeout)
+
+      setTimeout(() => {
+        ws.close()
+        resolve(null)
+      }, 1000)
+    })
+
+    ws.on('error', (err: unknown) => {
+      clearTimeout(timeout)
+      reject(err)
+    })
+
+    ws.on('message', (data: string) => {
+      assert.ok(data, 'Received log message from websocket')
```

Comment:
please validate the log lines.  Also, this test does not validate anything, as it can close without any messages being received.

Reply:
I am not sure we can easily mock this call `api.getRuntimeLiveLogsStream(pid)`

If we use `proxyquire`, we should mock the `getServer` helper, which calls `buildServer`, but leverage `@platformatic/service` to read the config and return a `fastify` instance. Therefore, I don't see how to mock `RuntimeApiClient` from `@platformatic/control`, which is the external method that should emit the logs data... 🤔

Reply:
for instance, on `web/backend/test/utils/calc.test.ts` we can more easily mock it, since we have direct access to the utils method:
```
const { calculateMetrics: calculateSingleWorker } = proxyquire(calcPath, {
    '@platformatic/control': {
      RuntimeApiClient: SingleWorkerClient
    }
  })
``` 

but in our case, we want to test the full `ws` through the `fastify` endpoint... so it's way more complicated to mock...

Reply:
What I can do is:
- Keep the barebone test (on the `/runtimes/:pid/logs/ws` URL) to check it doesn't return any error
- Move the inner websocket logic to a standalone function, where I can more easily mock stuff through `proxyquire` (or dependency injection)

@mcollina WDYT?

Reply:
why do you need to mock things at all?

Reply:
To simulate in the test to receive some logs, I need to mock the stream of data from @platformatic/control

Do you think there is an easier alternative?

Reply:
my bad, I was wrongly instantiating the websocket (it's receiving logs from the same `watt-admin` instance of the test). updated on 37bd0bf